datumaro/util/mask_tools.py

```diff
@@ -114,8 +114,8 @@ def make_binary_mask(mask):
     return mask.astype(bool)
 
 def bgr2index(img):
-    if img.dtype.kind not in {'b', 'i', 'u'}:
-        img = img.astype(np.uint8)
+    if img.dtype.kind not in {'b', 'i', 'u'} or img.dtype.itemsize < 4:
+        img = img.astype(np.uint32)
```

Comment:
Please check the calling contexts if they relied on number range narrowing from the previous conversion.

Reply:
I found only one context in which this function is called: function `bgr2index` is called in only one place to convert the bgr mask to the COCO index representation, and after this conversion, the calling context does not depend on the narrowing of the range:
https://github.com/openvinotoolkit/datumaro/blob/e8371e917a525d3026b547c717a1c2dfacd0a921/datumaro/plugins/coco_format/extractor.py#L187-L191
In the code above, we can see that the `mask` is redefined by the return value of the `bgr2index` function. But the `bgr2index` function can return a mask with values greater than 255, so that means we have to expand the mask's value type.